### PR TITLE
Initial Kura API.

### PR DESCRIPTION
**Pull request by humb1t**
Signed-off-by: Nikita Puzankov <humb1t@yandex.ru>

### CODE REVIEW

File: src/storage/kura/mod.rs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+/// Representation of a consistent storage.
+struct Disk {}
+
+impl Disk {
+    fn write(&mut self, block: iroha::Block) -> Result<u64, String> {
+        Result::Err("Not implemented yet.".to_string())
+    }
+}
+
+mod iroha {
```
Comment (takemiyamakoto) [inline src/storage/kura/mod.rs]: I would make a /model module and put this there.

File: src/storage/kura/model.rs [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct Signature {}
+
+/// A command is an intention to change the state of the network.
+/// For example, in order to create a new role in Iroha you have to issue Create role command.
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct Command {
+    version: u8,
+    command_type: u8,
```
Comment (eadventurous) [inline src/storage/kura/model.rs]: What is the mapping between `command_type` and actual commands like `AddAssetQuantity`? Maybe it's better to use **enum** here?

Comment (humb1t) [reply src/storage/kura/model.rs]: @eadventurous - good point.
Unfortunately for us and good for clients they can create custom commands and we will need to support them.

Comment (eadventurous) [reply src/storage/kura/model.rs]: Yes, I remember that requirement, in that case maybe something like:
```
pub enum Command {
    AddAssetQuantity(commands::AddAssetQuantity),
    ...
    Custom(Vec<u8>)
}
```

Comment (humb1t) [reply src/storage/kura/model.rs]: Let's think about it. We need to find cases where enum will be helpful when we work with commands.

File: src/storage/kura/mod.rs [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        }
+    }
+}
+
+fn merge_accounts_assets(
+    origin: CHashMap<String, Vec<model::Asset>>,
+    block: model::Block,
+) -> CHashMap<String, Vec<model::Asset>> {
+    origin.insert(
+        block.transactions[0].account_id.clone(),
```
Comment (takemiyamakoto) [inline src/storage/kura/mod.rs]: I'm not sure I understand the logic here. Why transactions[0]? Also, won't this fail if there are 0 transactions?

File: src/storage/kura/mod.rs [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+            id: "asset@domain".to_string(),
+        }],
+    );
+    origin
+}
+fn merge_inbound_transactions(
+    origin: CHashMap<String, Vec<model::Transaction>>,
+    block: model::Block,
+) -> CHashMap<String, Vec<model::Transaction>> {
+    origin
```
Comment (takemiyamakoto) [inline src/storage/kura/mod.rs]: Is this a WIP? If so, maybe add a //TODO?

**Pull request merged**